pages/profile/v2/[userId].tsx

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+import { useQuery } from "@apollo/client";
+import { useDispatch, useSelector } from "react-redux";
+import ExpandableContainer from "../../../components/coding/ExpandableContainer";
+import GoalsSectionComponent from "../../../components/coding/GoalsSectionComponent";
+import ProfileHeaderComponent from "../../../components/coding/profileV2/ProfileHeaderComponent";
+import SkillRatingsComponent from "../../../components/coding/SkillRatingsComponent";
+import {
+  FetchUserGoalsDataResponse,
+  FETCH_USER_GOALS,
+} from "../../../graphql/fetchUserGoals";
+import { useAuth } from "../../../lib/authContext";
+import { userGoalsSelector, setUserGoals } from "../../../redux/userGoalsSlice";
+import ProjectsSection from "../../../components/coding/ProjectsSection";
+import AchievementComponent from "../../../components/coding/profileV2/achievement_components/AchievementComponent";
+import { useRouter } from "next/router";
+import {
+  FetchTotalBadgesCountResponse,
+  FETCH_TOTAL_USER_BADGES_COUNT,
+} from "../../../graphql/fetchTotalUserBadgesCount";
+import {
+  FetchUserBadgesCountResponse,
+  FETCH_USER_BADGES_COUNT,
+} from "../../../graphql/fetchUserBadgesCount";
+import {
+  FetchUserProfileDataResponse,
+  FETCH_USER_PROFILE_DATA,
+} from "../../../graphql/fetchUserProfile";
+import {
+  profileSelector,
+  setUserProfile,
+  setUserBadgeCount,
+  setTotalBadgeCount,
+} from "../../../redux/profileSlice";
+
+type InternalProfileProps = {
+  userIdFromLink?: string;
+};
+
+export default function InternalProfile({
+  userIdFromLink,
+}: InternalProfileProps) {
+  const { user } = useAuth();
+  const router = useRouter();
+  const dispatch = useDispatch();
+
+  let userId;
+
+  if (userIdFromLink) {
```

Comment:
nice, maybe a comment here would be useful.

you could also do `const userId = userIdFromLink ? userIdFromLink : router.query.userId`

this way you can have a const variable instead.